src/daily_topic_newsletter.py

```
@@ -0,0 +1,160 @@
+        data = [item.__dict__ for item in self.history[-100:]]
+        with open(self.embed_store, "w", encoding="utf-8") as f:
+            json.dump(data, f)
+
+    # ------------------------------------------------------------------
+    def _embed_text(self, text: str) -> List[float]:
+        resp = openai.Embedding.create(model="text-embedding-ada-002", input=text)
+        return resp["data"][0]["embedding"]
+
+    # ------------------------------------------------------------------
+    def _cosine_similarity(self, v1: List[float], v2: List[float]) -> float:
+        dot = sum(a * b for a, b in zip(v1, v2))
```

Ruff (0.11.9)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)